reproman/resource/aws_condor.py

```diff
@@ -0,0 +1,180 @@
+            node.connect()
+            self.nodes.append(node)
+
+    def create(self):
+        """
+        Create a cluster of EC2 instances.
+
+        Yields
+        -------
+        dict : config and state parameters to capture in the inventory file
+        """
+        inventory = {
+            "type": "aws-condor",
+            "name": self.name,
+            "id": Resource._generate_id(),
+            "access_key_id": self.access_key_id,
+            "secret_access_key": self.secret_access_key,
+            "instance_type": self.instance_type,
+            "security_group": self.security_group,
+            "region_name": self.region_name,
+            "key_name": self.key_name,
+            "key_filename": self.key_filename,
+            "image": self.image,
+            "nodes": []
+        }
+        for i, node in enumerate(self.nodes):
+            inventory["nodes"].append({
+                "type": "aws-ec2",
+                "name": "{}_{}".format(self.name, i)
+            })
+        yield inventory
+
+        if not self.size:
+            raise ValueError("Missing --size switch when creating cluster")
+
+        def create_ec2(node):
+            node_inventory = {}
+            for resource_attrs in node.create():
+                node_inventory.update(resource_attrs)
+            return node_inventory
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            for i, node_inventory in enumerate(executor.map(create_ec2, self.nodes)):
+                inventory["nodes"][i].update(node_inventory)
+                yield inventory
+
+        lgr.info("Cluster %s is up and running!", self.name)
+
+        for i, node in enumerate(self.nodes):
+            if not node._ec2_instance.public_ip_address:
+                node._ec2_instance = node._ec2_resource.Instance(node.id)
+            condor_config = Template(
+                central_manager_ip=self.nodes[0]._ec2_instance.private_ip_address,
+                is_central_manager=(i == 0),
+                worker_nodes=self.nodes[1:]
+            ).render_cluster("condor_config.local.j2")
+            session = node.get_session()
+            session.execute_command(["sudo", "chmod", "0777", "/etc/condor/config.d"])
+            session.put_text(condor_config, "/etc/condor/config.d/00-nitrcce-cluster")
+            session.execute_command(["sudo", "rm", "/etc/condor/config.d/00-minicondor"])
+            session.execute_command(["sudo", "systemctl", "restart", "condor"])
+
+        yield {
+            'status': "Running"
+        }
+
+    def delete(self):
+        """
+        Terminate all the EC2 instances in the cluster.
+        """
+        for node in self.nodes:
+            node.delete()
+
+    def start(self):
+        """
+        Start all the EC2 instances in the cluster.
+        """
+        for node in self.nodes:
+            node.start()
+
+    def stop(self):
+        """
+        Stop all the EC2 instances in the AWS subscription.
```

Comment:
s/in the AWS subscription/in the cluster/ for consistency with the two docstrings above?